iOS/UI/Updates/MangaUpdatesView.swift

```diff
@@ -0,0 +1,188 @@
+                }
+            } header: {
+                Text(makeRelativeDate(days))
+                    .foregroundStyle(.primary)
+                    .font(.system(size: 16, weight: .medium))
+            }
+        }
+    }
+
+    var listItems: some View {
+        ForEach(entries.indices, id: \.self) { days in
+            Text(makeRelativeDate(days))
+                .foregroundColor(.primary)
+                .font(.system(size: 16, weight: .medium))
+
+            ForEach(entries[days].1) { mangaUpdate in
+                NavigationLink(destination: MangaView(manga: mangaUpdate.manga)) {
+                    MangaUpdateItemView(item: mangaUpdate)
+                }
+            }
+        }
+    }
+
+    private func loadNewEntries() async {
+        let mangaUpdates = await CoreDataManager.shared.container.performBackgroundTask { context in
+            CoreDataManager.shared.getRecentMangaUpdates(limit: 15, offset: offset, context: context).compactMap {
+                if let mangaObj = CoreDataManager.shared.getManga(
+                    sourceId: $0.sourceId,
+                    mangaId: $0.mangaId,
+                    context: context
+                ) {
+                    return MangaUpdateInfo(
+                        id: $0.id,
+                        sourceId: $0.sourceId,
+                        chapterId: $0.chapterId,
+                        mangaId: $0.mangaId,
+                        date: $0.date,
+                        manga: mangaObj.toManga(),
+                        chapter: $0.chapter?.toChapter()
+                    )
+                } else {
+                    return nil
+                }
+            }
+        }
+        if mangaUpdates.isEmpty {
+            await MainActor.run {
+                self.reachedEnd = true
+                self.loadingMore = false
+            }
+            return
+        }
+        var updatesDict: [Int: [MangaUpdateInfo]] = entries.reduce(into: [:]) { $0[$1.0] = $1.1 }
+        for obj in mangaUpdates {
+            let day = Calendar.autoupdatingCurrent.dateComponents(
+                Set([Calendar.Component.day]),
+                from: obj.date,
+                to: Date()
+            ).day ?? 0
+
+            var updatesOfTheDay = updatesDict[day] ?? []
+            updatesOfTheDay.append(obj)
+            updatesDict[day] = updatesOfTheDay
+        }
+        let finalUpdatesDict = updatesDict
+        await MainActor.run {
+            self.entries = finalUpdatesDict.map { ($0.key, $0.value) }.sorted { $0.0 < $1.0 }
+            self.offset += 15
+            self.loadingMore = false
+        }
+    }
+
+    private func reload() async {
+        if loadingMore {
+            loadingTask?.cancel()
+            loadingTask = nil
+        }
+        loadingMore = true
+        entries = []
+        offset = 0
+        reachedEnd = false
+        await loadNewEntries()
+    }
+
+    private func makeRelativeDate(_ days: Int) -> String {
```

Comment:
I copied this method from History, I think we should put it in a *Date extensions* or smth